### PR TITLE
Add initial Eclipse IDE description and Contribution instructions

### PR DESCRIPTION
**Pull request by HannesWell**
This is an early draft of the (work in progress) initial Eclipse IDE description and Contribution instructions.

Many parts are highly inspired by https://github.com/eclipse-platform/.github but are generalized to be applicable for all projects forming the Eclipse-IDE

### CODE REVIEW

File: CONTRIBUTING.md
Comment (opcoach) [inline CONTRIBUTING.md:1]: Regarding this sentence : It is also recommended to [add SSH public keys to your GitHub account](https://github.com/settings/keys).  I would say it is necessary as we should not use https:// do push on GitHub ...

We should also plan (later), to add a section on : 'from the intention to fix a bug to the PR proposal.'

File: profile/README.md
Comment (opcoach) [inline profile/README.md:1]: The section on technical documentation is rather scary... there should be an intermediate document to explain the whole process: from the intention to fix a bug to the PR proposal.  But we don't have it for the moment...

Comment (opcoach) [reply profile/README.md:1]: After our discussion today with @tfroment , as the visual is similar to eclipseide.org, it should be mentioned, in this README.md : 'This is the home page for the contributors' ... if you want only download eclipse-ide go to eclipse-ide.org... 

And we should have at this level the map  to explain what the contributors can access here ..

File: CONTRIBUTING.md
Comment (opcoach) [reply CONTRIBUTING.md:1]: We are here only describing the code contributor journey ... we should also mention documentation, translation, reviews contributions...

Comment (opcoach) [reply CONTRIBUTING.md:1]: Contributing to Eclipse should be renamed as :  The code contributor journey

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Contributing to Eclipse
```
Comment (opcoach) [inline CONTRIBUTING.md:1]: Would change the title to : The code contributor journey

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Thanks for your interest in contributing to Eclipse. Contributions are more than welcome!
```
Comment (opcoach) [inline CONTRIBUTING.md]: What about this : 

Welcome to the start of an exciting journey as a contributor to the Eclipse IDE project! 🌟 Becoming a contributor is more than just writing code—it’s about joining a vibrant community, learning new skills, and making a real impact. Whether you’re looking to expand your skills or give back to open-source, you’re in the right place.

Let’s break down the steps you’ll need to follow to get started. Each step links to a detailed guide, so don’t worry if you’re new to contributing—this process is designed to help you succeed.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+same email address like your [Eclipse account](https://accounts.eclipse.org/).
+If your GitHub account was registered with a different address, you can [add your Eclipse
+email address to the account](https://github.com/settings/emails) instead.
+
+Also add your GitHub account name to your [Eclipse account](https://accounts.eclipse.org/) to enable automated management of access rights for Eclipse project members. 
+You can do this by logging into your Eclipse account, choosing ["Edit Profile"](https://accounts.eclipse.org/user/edit) and add your GitHub account name in the social media links section.
+
+It is also recommended to [add SSH public keys to your GitHub account](https://github.com/settings/keys).
+
+## Creating an Eclipse Development Environment
```
Comment (opcoach) [inline CONTRIBUTING.md]: I would separate : 
## Install the Eclipse installer 

--> General instruction to install eclipse installer only once... (whatever the project we will contribute to).

## Install your code environment for the code you want to work on
--> Simple instruction to use the provided project's setup file in the installer.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+To provision an Eclipse installation and workspace to work on Eclipse we recommend to use the Eclipse Installer powered by `Eclipse Oomph`.
+- https://wiki.eclipse.org/Eclipse_Platform_SDK_Provisioning
+
+#TODO: Make this a comprehensive guide to use Oomph with the 'Contribution' buttons.
+
+- Most projects provide a Contributing button similar to:
+[![Create Eclipse Development Environment for the Eclipse SDK](https://download.eclipse.org/oomph/www/setups/svg/Platform_SDK.svg)](https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse-platform/eclipse.platform.releng.aggregator/master/oomph/PlatformSDKConfiguration.setup&show=true "Click to open Eclipse-Installer Auto Launch or drag onto your running installer's title area")
+
```
Comment (opcoach) [inline CONTRIBUTING.md]: should add : 
## Make your code modifications. 

Now comes the fun part—working on the code! Once you’ve cloned the project, you can modify it, run it locally, and test your changes. Don’t worry, [we have a guide for that too](https://github.com/eclipse-ide/.github/TBW).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+## Submit a contribution for a project host at Github
+
+The workflow to submit a contribution for Projects hosted at github.com, which is the vast majority.
```
Comment (opcoach) [inline CONTRIBUTING.md]: Here we can not go in details about PR as some projects do not use PR... should add a generic comment like this : 

Once you’ve tested your changes, it’s time to submit them. Depending on the project, you’ll submit your patch via a Pull Request (PR) or GerritHub. [Here’s how to submit your contribution](link_to_submission_guide).

And another step : 

Your contribution doesn’t end when you submit your patch! You’ll need to follow your PR or GerritHub submission, respond to feedback, and improve your work as needed. [Learn how to manage this process](link_to_follow_up_guide).

File: CONTRIBUTING.md
Comment (opcoach) [reply CONTRIBUTING.md:1]: Here is a very straight forward idea : 

# The Contributor Journey 🚀

Welcome to the start of an exciting journey as a contributor to the Eclipse IDE project! 🌟 Becoming a contributor is more than just writing code—it's about joining a vibrant community, learning new skills, and making a real impact. Whether you're looking to expand your skills or give back to open-source, you're in the right place.

Let’s break down the steps you’ll need to follow to get started. Each step links to a detailed guide, so don't worry if you're new to contributing—this process is designed to help you succeed.

---

## Step 1: Create Your GitHub and Eclipse Account  

First things first, if you don’t have a GitHub account yet, you’ll need one. It’s your gateway to the open-source community. [Create your GitHub account](link_to_github_signup) now if you haven’t already.
You will also have to register on eclipse account and sign the ECA (link to ECA)

---

## Step 2: Understand the Projects

The Eclipse IDE is part of a larger ecosystem of open-source projects. Each project may have its own setup and rules, but don’t sweat it—we’ll guide you through. You can find a detailed list of all projects that compose the IDE  [here](link_to_project_overview). This page is just a quick overview of the contribution process in general.

---

## Step 3: Install the 'Eclipse Installer' 🌟

To contribute, you'll need to install the  Eclipse installer on your machine. Don’t worry, it’s easy! [Here’s how to set up Eclipse](link_to_eclipse_install_guide). This optimized and powerful installer will allow you to configure any setup for any project later.

---

## Step 4: Clone the Code You Want to Work On

Each project has its own codebase. To contribute, you’ll need to clone the project you want to help with. With the help of the Eclipse Installer you can do this easily. You can find the step-by-step guide [here](link_to_cloning_guide).

---

## Step 5: Make Your Modifications

Now comes the fun part—working on the code! Once you’ve cloned the project, you can modify it, run it locally, and test your changes. Don’t worry, [we have a guide for that too](link_to_testing_guide).

---

## Step 6: Submit Your Work 💡

Once you’ve tested your changes, it’s time to submit them. Depending on the project, you’ll submit your patch via a Pull Request (PR) or GerritHub. [Here’s how to submit your contribution](link_to_submission_guide).

---

## Step 7: Track and Improve Your Contribution

Your contribution doesn’t end when you submit your patch! You’ll need to follow your PR or GerritHub submission, respond to feedback, and improve your work as needed. [Learn how to manage this process](link_to_follow_up_guide).

---

## Step 8: Document Your Changes 📝

Contributing to Eclipse isn’t just about the code—it’s about making it accessible for others. That’s why documentation is essential. After your code is submitted, make sure to [document your patch](link_to_documentation_guide) so that others can easily understand your work.

---

## Contributor's Checklist ✅

To help you stay on track, we’ve created a **Contributor’s Checklist**. This is a handy tool to ensure you’ve covered all the bases before submitting your patch. Check it out [here](link_to_checklist).

---

## Self-Evaluation Table

To help you evaluate your knowledge and progress as a contributor, we’ve provided a **self-evaluation table**. It’s a great way to see where you stand and what areas you can focus on. [Access the self-evaluation table](link_to_evaluation_table).

---

**Remember:** Contributing to Eclipse is a learning process, and you’re not alone. The community is here to support you at every step. Ready to make your mark? Let’s get started!

---

File: profile/README.md
Comment (HannesWell) [reply profile/README.md:1]: > The section on technical documentation is rather scary...

This is just for reference and put at the very end to not scare initially. So should we keep it there or remove it?
I would be in favor of keeping it as a side/non-prominent reference.

> After our discussion today with @tfroment , as the visual is similar to eclipseide.org, it should be mentioned, in this README.md : 'This is the home page for the contributors' ... if you want only download eclipse-ide go to eclipse-ide.org...

Done.

File: CONTRIBUTING.md
Comment (HannesWell) [reply CONTRIBUTING.md:1]: > Regarding this sentence : It is also recommended to [add SSH public keys to your GitHub account](https://github.com/settings/keys). I would say it is necessary as we should not use https:// do push on GitHub ...

Done. In general one can also use personal access tokens for authentication, but I think that's a more advanced topic. Using SSH keys is usually simple and convenient.

> 
> We should also plan (later), to add a section on : 'from the intention to fix a bug to the PR proposal.'

I hope that the entire content of this repository serves this goal.😃

File: projects.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+TODO: Make this a comprehensive guide about how to find the desired project and provide an overview about all available projects/repositories
+
+// TODO: provide a list of all SimRel (or even all Eclipse projects), ideally something that's managed by the EF, but it should also help  // interested contributors to find the corresponding project.
+// TODO: tips to identify the Plug-in of interest?
+// - Alt+Shift+F1
+// - Repository references encoded in Plug-in MANIFEST.MF files?!
```
Comment (HannesWell) [inline projects.md]: My idea was to add the overview of Eclipse IDE projects here.
In general what you have started in the [page-projects](https://github.com/eclipse-ide/.github/tree/page-projects) branch.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+## Creating an Eclipse Development Environment
+
+To provision an Eclipse installation and workspace to work on Eclipse we recommend to use the Eclipse Installer powered by `Eclipse Oomph`.
+
+Most projects provide an _Oomph setup button_ promiently on their main page, similar to:
+[![Create Eclipse Development Environment for Eclipse Platform](https://download.eclipse.org/oomph/www/setups/svg/Eclipse_Platform.svg)](
+https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse-platform/eclipse.platform/master/releng/org.eclipse.platform.setup/PlatformConfiguration.setup&show=true
+"Click to open Eclipse-Installer Auto Launch or drag into your running installer")
+
+The [`Provisioning tutorial`](https://wiki.eclipse.org/Eclipse_Platform_SDK_Provisioning) explains how to setup the full Eclipse development environment automatically using the _Oomph setup button_ of the desired project.
```
Comment (HannesWell) [inline CONTRIBUTING.md]: We could either add our step by step guide about how to use the Oomph configuration here or in another sub-page if you think it becomes too long otherwise?

File: CONTRIBUTING.md
Comment (opcoach) [inline CONTRIBUTING.md:1]: A general comment on this file...  As this is the top level file to explain how to contribute, it should give all the steps so as to give a general overview of the contributor journey... So some steps are missing and we must separate the Installer installation of the code to work on ... I will try to detail this afternoon...

Comment (opcoach) [inline CONTRIBUTING.md:1]: In my opinion, we should give in the contribution.md ALL the information of the journey in term of 'steps', from the registration to the documentation and up to the 'honor' and recognition...  The idea is not to describe like in other contribution pages what are the steps in detail in a very long document... For me the current proposed page is just like the page in other projects... too complicated in term of overview, not in term of explanation. For instance the GitHub/eclipse registration explanation is clear and simple and could stand in a sub page.

We have the chance to give an exciting overview of the contributor journey for new contributors. 

When we print the contributor page it could stand on 2 pages recto-verso like a map of the expected work

This page should give some easy steps that drives to a first success : install, change and run ... This is the first step for a contributor to be confident with the system. 

It should not contain all the GitHub and eclipse registration process in detail ... it should only mention it. When it is done, we will never read this again... (and this is optional unless we want to push a patch... )...Actually this paragraph could be mentioned before the first contribution...
  
The technical tuning and configuration should be written either in expandable paragraph (but with markdown I guess it is not possible), or with sub pages that will their own life cycle... For instance, we could have : 
 
 * Install the Eclipse Installer -> click here to download it, and launch it       (easy step, 5 mn). We explain it is only downloaded once, whatever the project, and will be automatically updated. 

 * Install a project to be changed :  (in this paragraph we have a simple screenshot of the main toolbar of the installer where we can paste the setup URL, and we give here an example of a setup (platform ui for instance), that will be installed. No more information, and a link to a page to learn the advanced features of the installer.  (After 5 more minutes, Eclipse for committer is operational with a project installed).

* Launch another instance : in this paragraph, we just give a way to change something, launch and see it is done.  This is motivating for developper to see that in 10 mn max, they have a stable environment to start their patch. 

This is the first part regarding install Tooling and with this a contributor can start to play ... 

Then we can explain the contribution phase, with bug fixing, registration to github/eclipse, review, .....

* Register to Eclipse and GitHub (exactly your paragraph with a link to a page that will explain the steps).

* Fix the bug, and contribute...  in this paragraph we can not detail too much, but we give links to resources to help and it is depending on the projects (debug, architecture documentation, ...). This is were we should mention the project description page. 

* Follow your review :  2 words about it and links to detailed pages... 

* Be recognized with its first contribution  :  explain the consequence of this patch... where the name of the contributor will be displayed, how he will be recognized. ...  


What I would like is to read this page, apply the basic steps and show that, without browsing tons of pages and comments, we have an instance of Eclipse that is runnable and modifiable, and ready to contribute.  The contributor will have all its life to read the detailed documentation. 

This is why, in my first comment, I proposed the different steps as mentioned in this comment... 

Basically the text written here is good, but should be moved in sub pages.

File: CONTRIBUTING.md (hunk 7 above)
Comment (opcoach) [reply CONTRIBUTING.md]: yes sure ! installer details should be explained in detail but not here

**Comment by HannesWell** [comment]
I have now incorporated all the changes we have discussed for a version zero. Incremental improvements will be built on top of this.
